src/ray/raylet/local_object_manager.h

```diff
@@ -68,12 +68,13 @@ class LocalObjectManager {
   /// Pin objects.
   ///
   /// \param object_ids The objects to be pinned.
-  /// \param objects Pointers to the objects to be pinned. The pointer should
-  /// be kept in scope until the object can be released.
   /// \param owner_address The owner of the objects to be pinned.
+  /// \param objects Pointers to the objects to be pinned. The pointer should
+  /// be kept in scope until the object can be released. If it's empty, it must
+  /// have already been pinned.
   void PinObjects(const std::vector<ObjectID> &object_ids,
-                  std::vector<std::unique_ptr<RayObject>> &&objects,
-                  const rpc::Address &owner_address);
+                  const rpc::Address &owner_address,
+                  std::vector<std::unique_ptr<RayObject>> objects = {});
```

Comment:
super nits: would an absl::optional<std::vector<std::unique_ptr<RayObject>>> make more sense?

Reply:
Btw, this doesn't have to be a vector (we only pin object one by one afaik)

Reply:
@rkooo567  I think later with ownership transfer we might need this one. I can revisit your comment later.

Reply:
@scv119 I kind of getting the feeling that we should use optional with simple types. I might be wrong. I'll update it to optional :)